### PR TITLE
Convert project to Maven.

### LINKED ISSUE #3: Convert project to use Maven instead of Gradle to align with the dapr java-sdk
Dapr's Java SDK use maven. Let's avoid the cognitive load of using 2 distinct build systems and use Maven here too.

### PR DESCRIPTION
**Pull request by tmacam**
This PR transitions this project to Maven, the same build tool used by
Dapr Java SDK. This alignment will reduce the number of tools used
by Java repositories in the project and allow for more code re-use.

We define only the major scaffolding for the project, re-using as
much configuration from Dapr Java SDK as sensible. Future PRs will
target the auto-generated from protocol buffers and the SDK proper.

We also moving the required JDK to JDK 17 - which should reduce the need
to add Lombok to the project.

Closes #3 

Signed-off-by: Tiago Alves Macambira <tmacam@burocrata.org>

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -29,261 +29,261 @@
...
+  </module>
 
-    <module name="RegexpHeader">
-        <property name="fileExtensions" value="java"/>
-        <property name="headerFile" value="${config_loc}/.java_header"/>
-    </module>
+  <module name="RegexpHeader">
+    <property name="fileExtensions" value="java"/>
+    <property name="headerFile" value=".java_header"/>
+  </module>
```
Comment (tmacam) [inline checkstyle.xml:57]: Effectively, what happened with this file is that it was overwritten with the original checksyle.xml from dapr/java-sdk, which doesn't need to define `config_loc`.

BTW, if "Hide whitespace" is selected in GitHub diff view, only these lines are shown in the diff for this file.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
+# Contribution Guidelines
+
+Thank you for your interest in Dapr!
+
+This project welcomes contributions and suggestions. Most contributions require you to signoff on your commits via 
+the Developer Certificate of Origin (DCO). When you submit a pull request, a DCO-bot will automatically determine 
+whether you need to provide signoff for your commit. Please follow the instructions provided by DCO-bot, as pull 
+requests cannot be merged until the author(s) have provided signoff to fulfill the DCO requirement.
+You may find more information on the DCO requirements [below](#developer-certificate-of-origin-signing-your-work).
```
Comment (tmacam) [inline CONTRIBUTING.md:9]: Copied over from dapr/java-sdk

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    <developers>
+        <developer>
+            <name>Dapr</name>
+            <email>daprweb@microsoft.com</email>
+            <organization>Dapr</organization>
+            <organizationUrl>https://dapr.io</organizationUrl>
+        </developer>
+    </developers>
+    
+    <scm>
```
Comment (philliphoff) [inline pom.xml:158]: Should this refer to the `dapr-sandbox/components-java-sdk`?

**Review by philliphoff — COMMENTED** [review]
Just one question re: SCM location for `pom.xml`.

File: pom.xml (hunk 3 above)
Comment (tmacam) [reply pom.xml:158]: Yes! You are absolutely right! Bad copy and paste :( Thanks for catching this.

**Pull request merged**